### PR TITLE
Adds a new gimmick trap, a springloaded boxing glove. Sprite included.

### PR DESCRIPTION
**Pull request by Bartimeus973**
## About the PR 
A simple trap, created by using a boxing glove on a mousetrap. When armed and in a container, a gift or someone's pocket, will trigger and punch whoever tries to take it out. The punch stuns for five seconds, play a ringing sound and sends the player flying in the opposite direction.

punching

## Why's this needed? 
A simple fun trap for clowns or staff assistants! Can be used in a variety of ways for example sending yourself into silly situations, pranking co-workers and adding some more cartoony levity to clown shenanigans.

Im not too used to github, so there's likely a few errors.

### CODE REVIEW

**Comment by ThatFiveGuys** [comment]
5 second stun seem way too strong for combat especially for things like stealth storages, personally id say remove the stun entirely. Also would be worthwhile testing multiple of these traps in a box to see if they stack.

File: code/obj/item/mousetrap.dm [hunk 1]
```diff
@@ -483,3 +491,106 @@
...
+
+				H.throw_unlimited = 1
+				H.throw_at(curr, 8, 3)
+			..()
+
+		src.icon_state = "boxing_trap"
+		src.armed = 0
+
+		clear_armer()
+		return
```
Comment (Zonespace27) [inline code/obj/item/mousetrap.dm:596]:
```suggestion
```
see my comments on other returns

File: code/obj/item/mousetrap.dm [hunk 2]
```diff
@@ -200,6 +201,13 @@
 			src.buttbomb = BB
 			src.overlays += image('icons/obj/items/weapons.dmi', "trap-buttbomb")
 
+		else if (istype(C, /obj/item/clothing/gloves/boxing) && !src.grenade && !src.grenade_old && !src.pipebomb  && !src.signaler && !src.butt && !src.buttbomb)
+			var/obj/item/clothing/gloves/boxing/G = C
+			user.u_equip(G)
+			user.show_text("You crudely stitch together an improvised trap with some spare metal. This doesnt look like a good idea.", "blue")
```
Comment (Zonespace27) [inline code/obj/item/mousetrap.dm:207]:
```suggestion
			user.show_text("You crudely stitch together an improvised trap with some spare metal. This doesn't look like a good idea.", "blue")
```

**Review by Zonespace27 — COMMENTED** [review]
Nice to see you contributing! Nothing on balance here, just code

File: code/obj/item/mousetrap.dm [hunk 3]
```diff
@@ -483,3 +491,104 @@
 		if (src.mousetrap.buttbomb && src.armed)
 			playsound(src, 'sound/voice/farts/poo2.ogg', 30, 0, 0, 1.8)
 		..()
+
+// A gimmick trap for unsuspecting people
+/obj/item/mousepunch
```
Comment (Zonespace27) [inline code/obj/item/mousetrap.dm:496]: this probably go with being a child of `/obj/item/device`, the less direct childs of `/obj/item`, the better, usually.

File: code/obj/item/mousetrap.dm [hunk 4]
```diff
@@ -483,3 +491,104 @@
...
+
+				for(var/i=0, i<7, i++)
+					curr = get_step(curr, turn(H.dir, 180))
+
+				H.throw_unlimited = 1
+				H.throw_at(curr, 8, 3)
+			..()
+
+		src.icon_state = "boxing_trap"
+		src.armed = 0
```
Comment (Zonespace27) [inline code/obj/item/mousetrap.dm:592]:
```suggestion
		src.armed = FALSE
```

File: code/datums/controllers/action_controls.dm [hunk 5]
```diff
@@ -915,6 +915,12 @@ var/datum/action_controller/actions
 						for (var/mob/O in AIviewers(owner))
 							O.show_message("<span class='alert'><B>...and triggers it accidentally!</B></span>", 1)
 						MT.triggered(source, source.hand ? "l_hand" : "r_hand")
+				else if (istype(I, /obj/item/device/mousepunch))
+					var/obj/item/device/mousepunch/MT = I
+					if (MT.armed)
+						for (var/mob/O in AIviewers(owner))
+							O.show_message("<span class='alert'><B>...and triggers it accidentally!</B></span>", 1)
```
Comment (pali6) [inline code/datums/controllers/action_controls.dm:922]: Probably use visible_message instead

File: code/obj/item/device/mousepunch.dm [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	desc = "A boxing glove shoddily tied to a mousetrap, eager to meet some clumsy assistant's face."
+	w_class = W_CLASS_TINY
+	force = 0
+	throwforce = null
+	var/armed = FALSE
+	stamina_damage = 0
+	stamina_cost = 0
+	stamina_crit_chance = 5
+	var/mob/armer = null
+	event_handler_flags = USE_FLUID_ENTER
```
Comment (pali6) [inline code/obj/item/device/mousepunch.dm:16]: Why?

File: code/obj/item/device/mousepunch.dm [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+	armed
+		icon_state = "boxing_trap_armed"
+		armed = 1
+
+	examine()
```
Comment (pali6) [inline code/obj/item/device/mousepunch.dm:22]: Override get_desc instead

**Review by pali6 — CHANGES_REQUESTED** [review]

File: code/datums/controllers/action_controls.dm [hunk 8]
```diff
@@ -915,6 +915,12 @@ var/datum/action_controller/actions
 						for (var/mob/O in AIviewers(owner))
 							O.show_message("<span class='alert'><B>...and triggers it accidentally!</B></span>", 1)
 						MT.triggered(source, source.hand ? "l_hand" : "r_hand")
+				else if (istype(I, /obj/item/device/mousepunch))
+					var/obj/item/device/mousepunch/MT = I
+					if (MT.armed)
+						for (var/mob/O in AIviewers(owner))
```
Comment (pali6) [inline code/datums/controllers/action_controls.dm:921]: The visible_message should only be called once on the source of the message, it does this loop inside of itself.

**Review by pali6 — CHANGES_REQUESTED** [review]

File: code/datums/controllers/action_controls.dm (hunk 8 above)
Comment (Bartimeus973) [reply code/datums/controllers/action_controls.dm:921]: Setting it to src.show_message and src.owner.show_message appears to not cause the message to pop up.

Comment (pali6) [reply code/datums/controllers/action_controls.dm:921]: That's why I am telling you to use visible_message and not show_message here.

File: code/datums/controllers/action_controls.dm [hunk 9]
```diff
@@ -915,11 +915,16 @@ var/datum/action_controller/actions
...
+							O.visible_message("<span class='alert'><B>...and triggers it accidentally!</B></span>", 1)
+						MT.triggered(source, source.hand ? "l_hand" : "r_hand")
 				else if (istype(I, /obj/item/mine))
 					var/obj/item/mine/M = I
 					if (M.armed && M.used_up != 1)
-						for (var/mob/O in AIviewers(owner))
-							O.show_message("<span class='alert'><B>...and triggers it accidentally!</B></span>", 1)
+						source.show_message("<span class='alert'><B>...and triggers it accidentally!</B></span>", 1)
```
Comment (pali6) [inline code/datums/controllers/action_controls.dm:927]: Same here.

**Review by pali6 — CHANGES_REQUESTED** [review]

**Comment by pali6** [comment]
Do you plan on fixing the issues or can we let this close due to staleness?

**Comment by Kubius** [comment]
If you get back around to this post freeze, ping me in the discord and I can get it reopened.

**Pull request closed without merging**